core/src/main/java/hudson/model/Label.java

```diff
@@ -107,6 +108,16 @@ public int computeTotalExecutors() {
             public int computeQueueLength() {
                 return Jenkins.getInstance().getQueue().countBuildableItemsFor(Label.this);
             }
+
+            @Override
+            protected Set<Node> getNodes() {
+                return Label.this.getNodes();
+            }
+
+            @Override
+            protected boolean matches(SubTask item) {
+                return item.getAssignedLabel() == Label.this;
```

Comment:
This should probably be .equals(). Labels are not guaranteed to be the same instance even if they have the same name.